src/main/java/Parser.java

```diff
@@ -0,0 +1,127 @@
+import java.util.Scanner;
+
+public class Parser {
+    public static void handle(TaskList taskList, Ui ui) {
+        Scanner scanner = new Scanner(System.in);
+        boolean loop = true;
+        do {
+            try {
+                String input = scanner.nextLine();
+                String commandType = extractCommandType(input);
+                String[] inputSplit = input.split(" ", 2);
+
+                switch (commandType) {
+                    case "todo":
+                        taskList.addTodo(prepareTodo(inputSplit));
+                        break;
+                    case "deadline":
+                        taskList.addDeadline(prepareDeadline(inputSplit));
+                        break;
+                    case "event":
+                        taskList.addEvent(prepareEvent(inputSplit));
+                        break;
+                    case "done":
+                        taskList.markDone(prepareDoneDelete(inputSplit[1]));
+                        break;
+                    case "delete":
+                        taskList.deleteTask(prepareDoneDelete(inputSplit[1]));
+                        break;
+                    case "list":
+                        ui.listTasks(taskList);
+                        break;
+                    case "bye":
+                        ui.exitDuke();
+                        loop = false;
+                        break;
+                    default:
+                        throw new InvalidCommandException("");
+                    }
+            } catch (InvalidCommandException | InvalidTodoException
+                    | InvalidDeadlineException | InvalidEventException ex) {
+                System.out.println(ex);
+            }
```

Comment:
Perhaps you can group the respective exceptions as an InvalidTaskException? Then you won't have to refactor this line in the future.